e2e/app/page/notebook-cell.ts

```diff
@@ -34,17 +34,34 @@ export default class NotebookCell {
   }
 
   /**
-   * Set focus to (select) a notebook cell input.
+   * Set focus to (select) a notebook cell input. Retry up to 3 times if focus fails.
    * @returns ElementHandle to code input if exists.
    */
-  async focus(): Promise<ElementHandle> {
+  async focus(maxAttempts: number = 3): Promise<ElementHandle> {
+    const clickInCell = async (iframe: Frame): Promise<ElementHandle> => {
+      const selector = this.cellSelector(this.getCellIndex());
+      const cell = await iframe.waitForSelector(`${selector} .CodeMirror-code`, {visible: true});
+      await cell.click({delay: 10}); // focus
+      return cell;
+    };
+
+    const clickAndCheck = async (iframe: Frame): Promise<ElementHandle> => {
```

Comment:
ah, because I decided to ignore `body.notebook_app.edit_mode` is not found error after retries max was reached. let the test continue to run while log a warning so I can be informed about what happened. I need more time to learn and refine notebook functions after a lot of test runs.